### DISCUSSION TITLE
antd升级到5.20.0报编译错误

### DISCUSSION THREAD

**Discussion by liubing900623** [ant-design/ant-design#50242] [General]
![image](https://github.com/user-attachments/assets/281e1dd1-8d94-45b5-beff-dabe10610932)
ERROR in ./node_modules/@ant-design/fast-color/es/FastColor.js 50:9

**Comment by zombieJ** [comment]
看起来是你的 webpack 配置没有正确支持 Class Fields 解析。理论上是不用过 babel 的，如果过了可以搜一下 @babel/plugin-proposal-class-properties 配置。

**Comment by Samgao0312** [comment]
解决了吗？也遇到同样问题，不过我没有升级包的情况下遇到的，神奇。。。。

**Reply by liubing900623** [reply]
我也是没有升级，是antd升级了，npm install之后就报错了，之前也遇到过类似问题。

**Reply by liubing900623** [reply]
还没有解决，你解决了，帮忙说下。

**Comment by xulei890817** [comment]
一样，新拉起一个开发环境，遇到了这个问题

**Comment by tythia** [comment]
![20240806-162206](https://github.com/user-attachments/assets/9148b58b-f408-4fb7-9e83-49b57d8f83da)
一样问题，代码没更新，npm i就报这个错，antd@^5.2.2

**Comment by yoyo837** [comment]
Fixed by https://github.com/ant-design/fast-color/pull/53, 请清理lock和node_modules重装试试

**Reply by liubing900623** [reply]
清理了还是一样

**Reply by yinLiangDream** [reply]
不是你的语法问题，问题在于你们ts编译之后的产物是需要 @babel/plugin-proposal-class-properties 转编译，这个写法 webpack4 不支持，具体issue：https://github.com/webpack/webpack/issues/9708  ，要么将编译结果转为老版本的es5，要么我们后续的webpack配置自己都加一层plugin来处理你的产物。。。

**Reply by yinLiangDream** [reply]
其实就是你的这个产物的语法，webpack4 不支持
<img width="642" alt="image" src="https://github.com/user-attachments/assets/b243422e-25e6-4d52-ab41-bf7d8bdc09e1">

**Reply by liubing900623** [reply]
我们等更新还是升级到webpack5？

**Comment by vagusX** [comment]
具体用的 webpack 是哪个版本，可以发下版本号 @liubing900623 @Samgao0312

**Reply by liubing900623** [reply]
"webpack": "^4.0.0"

**Reply by yoyo837** [reply]
`npm ls webpack`一下看看具体版本

**Reply by vagusX** [reply]
看来 webpack 4 并不支持? https://github.com/webpack/webpack/issues/9708

<img src="https://github.com/user-attachments/assets/a47ece72-bdeb-401c-8c36-72d11dedc02e" width="200" />

**Reply by liubing900623** [reply]
![image](https://github.com/user-attachments/assets/43090469-1896-481f-b01d-b8ac2584e983)

**Reply by crazyair** [reply]
把项目上传上来吧，把业务代码删一删

**Reply by yinLiangDream** [reply]
> 看来 webpack 4 并不支持? [webpack/webpack#9708](https://github.com/webpack/webpack/issues/9708)
> 
> <img alt="" width="200" src="https://private-user-images.githubusercontent.com/6828924/355395474-a47ece72-bdeb-401c-8c36-72d11dedc02e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI5MzcyNjMsIm5iZiI6MTcyMjkzNjk2MywicGF0aCI6Ii82ODI4OTI0LzM1NTM5NTQ3NC1hNDdlY2U3Mi1iZGViLTQwMWMtOGMzNi03MmQxMWRlZGMwMmUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgwNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MDZUMDkzNjAzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YTM0MzY3Njc2NTNjMzMwMTkwZDQ2NTdlZWMwOTgxNWZjMzYyODcwMDFhM2M0MGFkZDIwZDRiMTBjN2ZjZWRlZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.rARb8QL6uO2jYl9SGy7M-52CA7g4-sgRtvnUTqZKsTE">

为啥产物要包含 
<img width="379" alt="image" src="https://github.com/user-attachments/assets/124e1846-51bf-4cb5-9f05-4e9fbcb0728c">
这种语法呢

**Comment by crazyair** [comment]
发版了，再试下

**Reply by liubing900623** [reply]
可以了，谢谢。
